app/models/manageiq/providers/google/resource.rb

```diff
@@ -0,0 +1,40 @@
+module ManageIQ::Providers::Google
+  class Resource
+    def initialize(uri)
+      @type = self.class.parse_resource_type(uri)
+    end
+
+    attr_reader :type
```

Comment:
Well the compute API looks to have around 35 different resource types:
https://cloud.google.com/compute/docs/reference/latest/

On second thought, it might be better to make `@type` private; it's not like we're exposing this class in an API. If anyone needs to understand a new type it is very obvious how to add it (and would be preferred they add it in as a method rather than code the .type check)

I'll make `@type` private (and uh, let's just leave the three resource types we have rather than add 35 of them!)